shell/platform/darwin/ios/framework/Source/FlutterView.mm

```diff
@@ -24,6 +25,7 @@
 @implementation FlutterView
 
 id<FlutterViewEngineDelegate> _delegate;
+std::shared_ptr<flutter::IOSGLContext> _onscreenContext;
```

Comment:
If you don't put these instance variables in a block in the implementation, they are static and not tied to the lifecycle of the FlutterView. I suspect this is why you have to reset it explicitly below. The side effect of this change is now that there can only be a single onscreen context for all FlutterViews in the process. Also, a new FlutterView will nuke the previous IOSGLContext.

Putting both these variables in a block and get rid of the dealloc (which is now redundant since cxx objects held by objc object will get collected automatically) should fix all issues.

Reply:
Thanks for this. I was a little confused as my understanding of Obj-C is that it should be in a `{}` block after the `@implementation` (which I believe is what you're proposing here?) but I opted to keep it the same as the `_delegate` declaration as I figured I was missing something. I will move both into the block correctly and get rid of the explicit `dealloc`.